src/sentry/models/rulesnooze.py

```diff
@@ -0,0 +1,32 @@
+from django.db import models
+from django.db.models import CheckConstraint, Q
+
+from sentry.db.models import FlexibleForeignKey, Model, region_silo_only_model, sane_repr
+from sentry.db.models.fields.hybrid_cloud_foreign_key import HybridCloudForeignKey
+
+
+@region_silo_only_model
+class RuleSnooze(Model):
+    """
+    Duration an issue alert or metric alert is snoozed for a user. Null `until` value means snoozed forever.
+    """
+
+    __include_in_export__ = True
+
+    user_id = HybridCloudForeignKey("sentry.User", unique=True, on_delete="CASCADE")
```

Comment:
Hmm, so I assume you want to have one row per user per rule? At the moment, this is saying "one row per user"

In that case, we need two uniques on:
 - (user_id, rule)
 - (user_id, alert_rule)


You can do this via https://docs.djangoproject.com/en/4.1/ref/models/options/#unique-together